### PR TITLE
chore: Enable build / unit tests

### PR DESCRIPTION
**Pull request by MSNev**

### CODE REVIEW

File: pkgs/api/package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "test": "npm run test:node && npm run test:browser && npm run test:webworker",
     "test:browser": "",
     "test:debug": "nyc karma start ./karma.debug.conf.js --wait",
-    "test:node": "nyc ts-mocha -p tsconfig.json test/**/*.test.ts --exclude 'test/browser/**/*.ts'",
+    "test:node": "nyc ts-mocha -p tsconfig.json 'test/**/*.test.ts' --exclude 'test/browser/**/*.ts'",
```
Comment (MSNev) [inline pkgs/api/package.json:34]: The node tests ran fine on windows, but not on ubuntu without the quotes

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        run: |
+          npm run rebuild --verbose
+
+      - name: Unit tests
+        run: |
+          # TODO(legendecas): webpack https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported
+          if [ "${{ matrix.node_version }}" = "18" ]; then
+            export NODE_OPTIONS=--openssl-legacy-provider
+          fi
+          npm run test --verbose
```
Comment (MSNev) [inline .github/workflows/unit-test.yml:42]: This runs ALL tests (node, browser and worker -- if configured) for all migrated components

File: .github/workflows/unit-test.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          npm install
+          node common/scripts/install-run-rush.js check && node common/scripts/install-run-rush.js update
+
+      - name: Build 🔧
+        run: |
+          npm run rebuild --verbose
+
+      - name: Unit tests
+        run: |
+          # TODO(legendecas): webpack https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported
```
Comment (martinkuba) [inline .github/workflows/unit-test.yml:38]: What is the todo?

File: .github/workflows/unit-test.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+          fi
+          npm run test --verbose
+
+  node-windows-tests:
+    runs-on: windows-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node_version: [ 16 ]
```
Comment (martinkuba) [inline .github/workflows/unit-test.yml:50]: Why only 16 on Windows?

Comment (MSNev) [reply .github/workflows/unit-test.yml:50]: I tried running with 14 and 18 and it failed -- the existing tests only run on node 16 as well https://github.com/open-telemetry/opentelemetry-js/blob/main/.github/workflows/unit-test.yml#L62-L72

File: .github/workflows/unit-test.yml (hunk 3 above)
Comment (MSNev) [reply .github/workflows/unit-test.yml:38]: Taken from JS repo as it failed without this setting
https://github.com/open-telemetry/opentelemetry-js/blob/main/.github/workflows/unit-test.yml#L54-L57

**Review by martinkuba — APPROVED** [review]

**Pull request merged**